### PR TITLE
More point lookup optimizations

### PR DESCRIPTION
**Pull request by max-hoffman**
Use new `IndexLookup.IsPointLookup` to avoid duplicate work building range tuples.

companion PR: https://github.com/dolthub/go-mysql-server/pull/1239

### CODE REVIEW

**Comment by max-hoffman** [comment]
#benchmark

**Comment by max-hoffman** [comment]
#benchmark

File: go/libraries/doltcore/sqle/index/index_lookup.go [hunk 1]
```diff
@@ -115,15 +116,63 @@ func NewRangePartitionIter(ctx *sql.Context, t DoltTableable, lookup sql.IndexLo
...
+		}
+		if err = PutField(ctx, ns, tb, j, v); err != nil {
+			return nil, err
+		}
+	}
+	tup := tb.BuildPermissive(sharePool)
+	return &pointPartitionIter{r: prolly.Range{Tup: tup, Desc: tb.Desc}}, nil
+}
+
+type pointPartitionIter struct {
```
Comment (andrew-wm-arthur) [inline go/libraries/doltcore/sqle/index/index_lookup.go:140]: for every secondary lookup in a join we go from `sql.IndexLookup` -> `sql.PartitionIter` -> `sql.Partiiton` -> `sql.RowIter`. Is there any way to collapse that indirection, even if we keep the existing interfaces? Specifically I'm wondering if `pointPartitionIter` and `pointPartition` could be merged into one type

**Review by andrew-wm-arthur — APPROVED** [review]
LGTM!

File: go/libraries/doltcore/sqle/index/index_lookup.go (hunk 1 above)
Comment (max-hoffman) [reply go/libraries/doltcore/sqle/index/index_lookup.go]: Good call, that squeezed an extra 5% out of the join bench.

**Pull request merged**